src/test/java/com/es/phoneshop/model/product/ArrayListProductDaoTest.java

```diff
@@ -16,6 +23,35 @@ public void setup() {
 
     @Test
     public void testFindProductsNoResults() {
-        assertTrue(productDao.findProducts().isEmpty());
+        Product product = productDao.save(new Product("test", "test", new BigDecimal(100), usd, 0, "https://raw.githubusercontent.com/andrewosipenko/phoneshop-ext-images/master/manufacturer/Samsung/Samsung%20Galaxy%20S.jpg"));
```

Comment:
when you test  DAO you should not rely on it's other methods
you should mock this method save() and call only .findProducts()

if your save() is broken you will have x2 failed tests: all for save() and tests where this save() is called before the test method